.github/workflows/ci.yml

```diff
@@ -1,6 +1,14 @@
 name: CI
 
-on: [push, pull_request, workflow_dispatch]
```

Comment:
Frankly, I'd rather keep it simple and as it is. I don't feel any urge to optimize anything here.